src/main/java/Duke.java

```diff
@@ -1,10 +1,153 @@
+            at += words.get(i);
+            at += " ";
+        }
+        at = at.substring(0, at.length()-1);
+        return new Event(description, at);
+    }
+    private static Deadline newDeadline(String userInput) {
+        ArrayList<String> words = new ArrayList<String>(Arrays.asList(userInput.split(" ")));
+        words.remove(0);
+        String description = "";
+        String by = "";
+        int byIndex = -1;
+        for (int i=0; i<words.size(); i++) {
+            if (words.get(i).equals("/by")) {
+                byIndex = i;
+            }
+        }
+        for (int i=0; i<byIndex; i++) {
+            description += words.get(i);
+            description += " ";
+        }
+        description = description.substring(0, description.length()-1);
+        for (int i=byIndex+1; i < words.size(); i++) {
+            by += words.get(i);
+            by += " ";
+        }
+        by = by.substring(0, by.length()-1);
+        return new Deadline(description, by);
+    }
+
+    //TODO edge case: fix what to print when user input doesn't contain /by or /at during deadline/event command
+    private static void addCommand(String userInput, String taskType) {
+        if (taskType == "todo") {
+            userList.add(newTodo(userInput));
+        } else if (taskType == "event") {
+            userList.add(newEvent(userInput));
+        } else if (taskType == "deadline") {
+            userList.add(newDeadline(userInput));
+        }
+        printAdded();
+    }
+
+    private static void printAdded() {
+        System.out.print(lines);
+        System.out.println("Got it . I've added this task:");
+        System.out.print("  ");
+        System.out.println(userList.get(userList.size()-1));
+        System.out.println("Now you have " + userList.size() + " tasks in the list.");
+        System.out.print(lines);
+    }
+
+    private static boolean containCommand(String userInput, String command) {
+        int myIntFlag = userInput.indexOf(command);
+        return myIntFlag == 0; // returns true only if first index == command
+    }
+    //Main bot control
+    private static void botResponse(String userInput) {
+        while (!containCommand(userInput, "bye")) {          // if user command is not bye, go inside loop
+            if (containCommand(userInput, "list")) {
+                listCommand();
+            } else if (containCommand(userInput, "done ")) {
+                doneCommand(userInput);
+            } else if (containCommand(userInput, "todo")){
+                addCommand(userInput, "todo");
+            } else if (containCommand(userInput, "event")) {
+                addCommand(userInput, "event");
+            } else if (containCommand(userInput, "deadline")) {
+                addCommand(userInput, "deadline");
+            }
+            userInput = in.nextLine();
+        }
+        System.out.print(lines + "Bye. Hope to see you again soon!\n" + lines);
```

Comment:
Would it be clearer to get the userInput in another method rather than putting it in your botResponse method?